onnxruntime/core/codegen/mti/tensor/gather_elements.cc

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/codegen/mti/tensor/gather_elements.h"
+
+#include "core/codegen/mti/mti_tvm_utils.h"
+#include <topi/transform.h>
+
+namespace onnxruntime {
+namespace tvm_codegen {
+
+tvm::Tensor GatherElements(const tvm::Tensor& t,
+                           int64_t axis,
+                           const tvm::Tensor& indices,
+                           const std::string& name) {
+  tvm::Array<tvm::Expr> output_shape;
+  int64_t indices_rank = static_cast<int64_t>(indices->shape.size());
+  // output shape is the same as indices
+  for (int64_t i = 0; i < indices_rank; ++i)
+    output_shape.push_back(indices->shape[i]);
+
+  tvm::Expr idx_upper_bound = t->shape[axis];
```

Comment:
>idx_upper_bound [](start = 12, length = 15)

make this dim -1, so we can save some compute in the loop below. And the upper_bound would be inclusive